actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -23,15 +24,16 @@ class ActionCable.Connection
       ActionCable.log("Attemped to open WebSocket, but existing socket is #{@getState()}")
       throw new Error("Existing connection must be closed before opening")
     else
-      ActionCable.log("Opening WebSocket, current state is #{@getState()}")
+      ActionCable.log("Opening WebSocket, current state is #{@getState()}, subprotocols: #{protocols}")
       @uninstallEventHandlers() if @webSocket?
-      @webSocket = new WebSocket(@consumer.url)
+      @webSocket = new WebSocket(@consumer.url, protocols)
       @installEventHandlers()
       @monitor.start()
       true
 
-  close: ->
-    @webSocket?.close()
+  close: ({allowReconnect} = {allowReconnect: true}) ->
+    @monitor.stop() unless allowReconnect
+    @webSocket?.close() unless @isClosed()
```

Comment:
Could do `@webSocket?.close() if @isActive()` and drop the `isClosed` method entirely. Side note: we intentionally named `isActive` to not match a connection state and `isClosed` doesn't follow that pattern.